libnmstate/prettystate.py

```diff
@@ -27,7 +27,10 @@ class PrettyState(object):
     def __init__(self, state):
 
         yaml.add_representer(OrderedDict, represent_ordereddict)
-        self.state = order_state(state)
+        if 'interfaces' in state:
```

Comment:
This is not that nice, it is depending on the caller: every new caller scenario will require changes here.
How about require the caller to add all it needs to add to make it here unified (add the `interfaces` key)?
Another option is to input in the constructor the order function, leaving the responsibility to the caller to choose the correct one.

Reply:
Initially I made it the responsibility of the caller as well, however `order_state()` requires the `interface` key to be present, therefore we can just check for it directly:

```
iface_states = state.pop('interfaces', [])
```

Therefore I would prefer to make it simplier for the caller for now.

Reply:
It is one `if` now but has the potential of increasing in the future (it will happen, `interfaces` are not the only root subtree we will have for sure). There is nothing common in something that in the entry point has a switch (multi if) statement.
Why not let the caller just add it? {'interfaces': ifaces_states}

If you have strong feeling on this point, another options is to just extract the `if` part out of the constructor to a factory function that will do exactly the same.